File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03849/p03849_num7_parsed.cpp
#include <iostream>
using namespace std;

const long long MOD = 1000000007;

long long countPairs(long long N) {
    long long result = 0;
    for (int u = 0; u <= N; u++) {
        for (int v = u; v <= N; v++) {
            long long a = (u + v) / 2;
            long long b = v - a;
            if ((a ^ b) == u) {
                result++;
            }
        }
    }
    return result % MOD;
}

int main() {
    long long N;
    cin >> N;
    
    long long ans = countPairs(N);
    cout << ans << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03849/p03849_num7_parsed.py
MOD = 10**9 + 7

def count_pairs(N):
    num_pairs = 0
    for u in range(N+1):
        for v in range(N+1):
            a = u ^ v
            b = v - a
            if a >= 0 and b >= 0 and a + b == v:
                num_pairs += 1
    
    return num_pairs % MOD

N = int(input())
result = count_pairs(N)
print(result)